rustler/src/types/mod.rs

```diff
@@ -126,3 +126,37 @@ where
         }
     }
 }
+
+impl<'a, K, V> Decoder<'a> for std::collections::HashMap<K, V>
+where
+    K: Decoder<'a> + Eq + std::hash::Hash,
+    V: Decoder<'a>,
+{
+    fn decode(term: Term<'a>) -> NifResult<Self> {
+        let it = MapIterator::new(term).ok_or_else(|| Error::BadArg)?;
+
+        let mut map = std::collections::HashMap::new();
```

Comment:
It might make sense to retrieve the map's size with `enif_get_map_size` and call `with_capacity()` here.

Reply:
d5a23a8